modules/sor/sorV2/lib/pools/weighted/weightedPool.ts

```diff
@@ -131,6 +131,18 @@ export class WeightedPool implements BasePool {
         return (tOut.amount * this.MAX_OUT_RATIO) / WAD;
     }
 
+    public getLimitAmountAddLiquidity(tokenIn: Token): bigint {
+        const tIn = this.tokenMap.get(tokenIn.wrapped);
+        if (!tIn) {
+            throw new Error('getLimitAmountAddLiquidity: Token not found');
+        }
+        return (tIn.amount * this.MAX_IN_RATIO) / WAD;
+    }
+
+    public getLimitAmountRemoveLiquidity(): bigint {
+        return (this.totalShares * this.MAX_IN_RATIO) / WAD;
```

Comment:
Also not sure if this number makes sense. Did you confirm this info?

Reply:
I didn't confirm this number, but I understood that this limit is necessary to prevent the invariant to be changed too much with swaps/adds/removes that exceeds 30% of the balance of the token to be added, Tom French made this implementation back in the old SOR, I'll confirm if this makes sense with the SOR guys

Reply:
I applied the logic of a swap with BPT for RemoveLiquidity, if a swap with BPT had the limit of 30% of the totalSupply, I thought the RemoveLiquidity with a single token out should have the same limit, but maybe I'm missing something here?

Reply:
I don't think it's as straightforward as that. If you take the stable pool swap limit for example, it's approximately the whole token balance. In the context of removeLiquidity, you wouldn't be able to remove liquidity to a single token with nearly all BPT balance.
As this is actually a limit stablished by the pool contract, I'd double check with SC team on what would be the limits for single token add/remove operations.
When asking them, don't forget that this is in the context of v3 only.

Reply:
Yeah, you're right, there is no limit in the V3, not even for swaps actually, I'll change the flow, so V3 pools will not have to calculate the limit, thanks

Reply:
Now that I'm making the tests for these limits, I saw that this remove liquidity limits are not possible, to remove all the amount of one of the tokens, the user would have to send the invariant of the pool to 0, it breaks the current math. Maybe keep the limit as 30% of the amount of tokenOut is a better option in the moment? I'll rollback to 30% and then we can change it in the future.